components/locid/src/subtags/language.rs

```diff
@@ -58,25 +58,34 @@ impl Language {
     ///
     /// assert_eq!(lang, "en");
     /// ```
-    pub fn from_bytes(v: &[u8]) -> Result<Self, ParserError> {
-        let slen = v.len();
-
-        if !LANGUAGE_LENGTH.contains(&slen) {
-            return Err(ParserError::InvalidLanguage);
-        }
+    pub const fn from_bytes(v: &[u8]) -> Result<Self, ParserError> {
+        Self::from_bytes_manual_slice(v, 0, v.len())
+    }
 
-        let s = TinyAsciiStr::from_bytes(v).map_err(|_| ParserError::InvalidLanguage)?;
+    /// Equivalent to [`from_bytes(bytes[start..end])`](Self::from_bytes),
+    /// but callable in a `const` context (which range indexing is not).
+    pub const fn from_bytes_manual_slice(
+        v: &[u8],
+        start: usize,
+        end: usize,
+    ) -> Result<Self, ParserError> {
+        let slen = end - start;
 
-        if !s.is_ascii_alphabetic() {
+        if slen < *LANGUAGE_LENGTH.start() || *LANGUAGE_LENGTH.end() < slen {
             return Err(ParserError::InvalidLanguage);
         }
 
-        let value = s.to_ascii_lowercase();
+        if slen == 3
+            && v[start] == UND_VALUE.all_bytes()[0]
+            && v[start + 1] == UND_VALUE.all_bytes()[1]
+            && v[start + 2] == UND_VALUE.all_bytes()[2]
+        {
```

Comment:
`question`: am I correct that this will not work for non-canonicalized case for `und`? Like `UND` will not return `None` here?

Reply:
I think this is a pre-existing bug, but may be worth fixing here.

Reply:
I probably introduced this when I moved the check before the lowercasing. However after the merge this is deleted.